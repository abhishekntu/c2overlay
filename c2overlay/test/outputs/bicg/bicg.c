/****************************************************************/
/*   chebyshev		                                        */
/****************************************************************/

#ifdef TEST
#include <stdio.h>
#endif

#include<math.h>

#ifdef ALGO
void foo()
{
    int A [9];
    int p [3];
    int q [3];
    int r [3];
    int s [3];
    {
        q[0] = A[0] * p[0] + A[1] * p[1] + A[2] * p[2];
        q[1] = A[3] * p[0] + A[4] * p[1] + A[5] * p[2];
        q[2] = A[6] * p[0] + A[7] * p[1] + A[8] * p[2];
    }
    {
        s[0] = A[0] * r[0] + A[3] * r[1] + A[6] * r[2];
        s[1] = A[1] * r[0] + A[4] * r[1] + A[7] * r[2];
        s[2] = A[2] * r[0] + A[5] * r[1] + A[8] * r[2];
    }

}
#endif


#ifdef TEST
int main(void)
{
	int x;
 	int result;
  	foo();
  	return 0;
}
#endif
