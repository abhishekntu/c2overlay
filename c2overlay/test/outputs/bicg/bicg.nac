procedure foo()
{
  localvar s32 D_2407;
  localvar s32 D_2408;
  localvar s32 D_2409;
  localvar s32 D_2410;
  localvar s32 D_2411;
  localvar s32 D_2412;
  localvar s32 D_2413;
  localvar s32 D_2414;
  localvar s32 D_2415;
  localvar s32 D_2416;
  localvar s32 D_2417;
  localvar s32 D_2418;
  localvar s32 D_2419;
  localvar s32 D_2420;
  localvar s32 D_2421;
  localvar s32 D_2422;
  localvar s32 D_2423;
  localvar s32 D_2424;
  localvar s32 D_2425;
  localvar s32 D_2426;
  localvar s32 D_2427;
  localvar s32 D_2428;
  localvar s32 D_2429;
  localvar s32 D_2430;
  localvar s32 D_2431;
  localvar s32 D_2432;
  localvar s32 D_2433;
  localvar s32 D_2434;
  localvar s32 D_2435;
  localvar s32 D_2436;
  localvar s32 D_2437;
  localvar s32 D_2438;
  localvar s32 D_2439;
  localvar s32 D_2440;
  localvar s32 D_2441;
  localvar s32 D_2442;
  localvar s32 D_2443;
  localvar s32 D_2444;
  localvar s32 D_2445;
  localvar s32 D_2446;
  localvar s32 D_2447;
  localvar s32 D_2448;
  localvar s32 D_2449;
  localvar s32 D_2450;
  localvar s32 D_2451;
  localvar s32 A[9];
  localvar s32 p[3];
  localvar s32 q[3];
  localvar s32 r[3];
  localvar s32 s[3];
  localvar s32 t00000;
  localvar s32 t00001;
  localvar s32 t00002;
  localvar s32 t00003;
  localvar s32 t00004;
  localvar s32 t00005;
  localvar s32 t00006;
  localvar s32 t00007;
  localvar s32 t00008;
  localvar s32 t00009;
  localvar s32 t00010;
  localvar s32 t00011;
  localvar s32 t00012;
  localvar s32 t00013;
  localvar s32 t00014;
  localvar s32 t00015;
  localvar s32 t00016;
  localvar s32 t00017;
  localvar s32 t00018;
  localvar s32 t00019;
  localvar s32 t00020;
  localvar s32 t00021;
  localvar s32 t00022;
  localvar s32 t00023;
  localvar s32 t00024;
  localvar s32 t00025;
  localvar s32 t00026;
  localvar s32 t00027;
  localvar s32 t00028;
  localvar s32 t00029;
  localvar s32 t00030;
  localvar s32 t00031;
  localvar s32 t00032;
  localvar s32 t00033;
  localvar s32 t00034;
  localvar s32 t00035;
  localvar s32 t00036;
  localvar s32 t00037;
  localvar s32 t00038;
  localvar s32 t00039;
  localvar s32 t00040;
  localvar s32 t00041;
L0005:
  D_2407 <= load A,0;
  D_2408 <= load p,0;
  D_2409 <= mul D_2407,D_2408;
  D_2410 <= load A,1;
  D_2411 <= load p,1;
  D_2412 <= mul D_2410,D_2411;
  D_2413 <= add D_2409,D_2412;
  D_2414 <= load A,2;
  D_2415 <= load p,2;
  D_2416 <= mul D_2414,D_2415;
  D_2417 <= add D_2413,D_2416;
  q <= store D_2417,0;
  D_2418 <= load A,3;
  D_2408 <= load p,0;
  D_2419 <= mul D_2418,D_2408;
  D_2420 <= load A,4;
  D_2411 <= load p,1;
  D_2421 <= mul D_2420,D_2411;
  D_2422 <= add D_2419,D_2421;
  D_2423 <= load A,5;
  D_2415 <= load p,2;
  D_2424 <= mul D_2423,D_2415;
  D_2425 <= add D_2422,D_2424;
  q <= store D_2425,1;
  D_2426 <= load A,6;
  D_2408 <= load p,0;
  D_2427 <= mul D_2426,D_2408;
  D_2428 <= load A,7;
  D_2411 <= load p,1;
  D_2429 <= mul D_2428,D_2411;
  D_2430 <= add D_2427,D_2429;
  D_2431 <= load A,8;
  D_2415 <= load p,2;
  D_2432 <= mul D_2431,D_2415;
  D_2433 <= add D_2430,D_2432;
  q <= store D_2433,2;
  D_2407 <= load A,0;
  D_2434 <= load r,0;
  D_2435 <= mul D_2407,D_2434;
  D_2418 <= load A,3;
  D_2436 <= load r,1;
  D_2437 <= mul D_2418,D_2436;
  D_2438 <= add D_2435,D_2437;
  D_2426 <= load A,6;
  D_2439 <= load r,2;
  D_2440 <= mul D_2426,D_2439;
  D_2441 <= add D_2438,D_2440;
  s <= store D_2441,0;
  D_2410 <= load A,1;
  D_2434 <= load r,0;
  D_2442 <= mul D_2410,D_2434;
  D_2420 <= load A,4;
  D_2436 <= load r,1;
  D_2443 <= mul D_2420,D_2436;
  D_2444 <= add D_2442,D_2443;
  D_2428 <= load A,7;
  D_2439 <= load r,2;
  D_2445 <= mul D_2428,D_2439;
  D_2446 <= add D_2444,D_2445;
  s <= store D_2446,1;
  D_2414 <= load A,2;
  D_2434 <= load r,0;
  D_2447 <= mul D_2414,D_2434;
  D_2423 <= load A,5;
  D_2436 <= load r,1;
  D_2448 <= mul D_2423,D_2436;
  D_2449 <= add D_2447,D_2448;
  D_2431 <= load A,8;
  D_2439 <= load r,2;
  D_2450 <= mul D_2431,D_2439;
  D_2451 <= add D_2449,D_2450;
  s <= store D_2451,2;
  L_EXIT <= jmpun;
L_EXIT:
  nop;
}
