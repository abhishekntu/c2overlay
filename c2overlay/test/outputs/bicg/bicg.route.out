Array size: 7 x 7 logic blocks.

Routing:

Net 0 (N8)

SOURCE (3,8)  Pad: 1  
  OPIN (3,8)  Pad: 1  
 CHANX (3,7)  Track: 3  
 CHANY (2,7)  Track: 3  
 CHANY (2,6)  Track: 3  
 CHANY (2,5)  Track: 3  
 CHANY (2,4)  Track: 3  
 CHANY (2,3)  Track: 3  
 CHANY (2,2)  Track: 3  
  IPIN (3,2)  Pin: 3  
  SINK (3,2)  Class: 0  
 CHANY (2,3)  Track: 3  
 CHANX (3,2)  Track: 2  
 CHANX (4,2)  Track: 2  
 CHANX (5,2)  Track: 2  
  IPIN (5,2)  Pin: 0  
  SINK (5,2)  Class: 0  
 CHANY (2,3)  Track: 3  
  IPIN (2,3)  Pin: 1  
  SINK (2,3)  Class: 0  


Net 1 (N9)

SOURCE (3,0)  Pad: 1  
  OPIN (3,0)  Pad: 1  
 CHANX (3,0)  Track: 3  
  IPIN (3,1)  Pin: 2  
  SINK (3,1)  Class: 0  
 CHANX (3,0)  Track: 3  
 CHANY (2,1)  Track: 2  
 CHANX (3,1)  Track: 2  
  IPIN (3,2)  Pin: 2  
  SINK (3,2)  Class: 0  


Net 2 (N1)

SOURCE (0,4)  Pad: 1  
  OPIN (0,4)  Pad: 1  
 CHANY (0,4)  Track: 3  
 CHANX (1,3)  Track: 2  
 CHANX (2,3)  Track: 2  
  IPIN (2,3)  Pin: 0  
  SINK (2,3)  Class: 0  
 CHANX (1,3)  Track: 2  
 CHANY (1,3)  Track: 3  
 CHANY (1,2)  Track: 3  
  IPIN (2,2)  Pin: 3  
  SINK (2,2)  Class: 0  


Net 3 (N2)

SOURCE (0,2)  Pad: 1  
  OPIN (0,2)  Pad: 1  
 CHANY (0,2)  Track: 2  
 CHANY (0,3)  Track: 2  
  IPIN (1,3)  Pin: 3  
  SINK (1,3)  Class: 0  
  OPIN (0,2)  Pad: 1  
 CHANY (0,2)  Track: 3  
 CHANY (0,1)  Track: 3  
  IPIN (1,1)  Pin: 3  
  SINK (1,1)  Class: 0  
 CHANY (0,1)  Track: 3  
 CHANX (1,0)  Track: 2  
 CHANX (2,0)  Track: 2  
 CHANX (3,0)  Track: 2  
 CHANX (4,0)  Track: 2  
 CHANX (5,0)  Track: 2  
  IPIN (5,1)  Pin: 2  
  SINK (5,1)  Class: 0  


Net 4 (N3)

SOURCE (2,0)  Pad: 1  
  OPIN (2,0)  Pad: 1  
 CHANX (2,0)  Track: 3  
  IPIN (2,1)  Pin: 2  
  SINK (2,1)  Class: 0  
  OPIN (2,0)  Pad: 1  
 CHANX (2,0)  Track: 0  
 CHANY (2,1)  Track: 0  
  IPIN (3,1)  Pin: 3  
  SINK (3,1)  Class: 0  
 CHANY (2,1)  Track: 0  
 CHANX (3,1)  Track: 0  
 CHANX (4,1)  Track: 0  
  IPIN (4,1)  Pin: 0  
  SINK (4,1)  Class: 0  


Net 5 (N4)

SOURCE (8,3)  Pad: 1  
  OPIN (8,3)  Pad: 1  
 CHANY (7,3)  Track: 3  
 CHANY (7,2)  Track: 3  
  IPIN (7,2)  Pin: 1  
  SINK (7,2)  Class: 0  
 CHANY (7,2)  Track: 3  
 CHANX (7,1)  Track: 3  
 CHANX (6,1)  Track: 3  
 CHANX (5,1)  Track: 3  
  IPIN (5,2)  Pin: 2  
  SINK (5,2)  Class: 0  


Net 6 (N5)

SOURCE (8,5)  Pad: 1  
  OPIN (8,5)  Pad: 1  
 CHANY (7,5)  Track: 1  
 CHANY (7,4)  Track: 1  
 CHANY (7,3)  Track: 1  
 CHANY (7,2)  Track: 1  
 CHANY (7,1)  Track: 1  
  IPIN (7,1)  Pin: 1  
  SINK (7,1)  Class: 0  
 CHANY (7,3)  Track: 1  
 CHANX (7,2)  Track: 1  
 CHANX (6,2)  Track: 1  
  IPIN (6,2)  Pin: 0  
  SINK (6,2)  Class: 0  


Net 7 (N6)

SOURCE (5,8)  Pad: 1  
  OPIN (5,8)  Pad: 1  
 CHANX (5,7)  Track: 3  
 CHANY (4,7)  Track: 3  
 CHANY (4,6)  Track: 3  
 CHANY (4,5)  Track: 3  
 CHANY (4,4)  Track: 3  
 CHANY (4,3)  Track: 3  
  IPIN (4,3)  Pin: 1  
  SINK (4,3)  Class: 0  
 CHANY (4,3)  Track: 3  
 CHANY (4,2)  Track: 3  
  IPIN (4,2)  Pin: 1  
  SINK (4,2)  Class: 0  
 CHANY (4,2)  Track: 3  
 CHANX (5,1)  Track: 2  
 CHANX (6,1)  Track: 2  
  IPIN (6,2)  Pin: 2  
  SINK (6,2)  Class: 0  


Net 8 (N7)

SOURCE (6,0)  Pad: 1  
  OPIN (6,0)  Pad: 1  
 CHANX (6,0)  Track: 3  
  IPIN (6,1)  Pin: 2  
  SINK (6,1)  Class: 0  
 CHANX (6,0)  Track: 3  
 CHANY (5,1)  Track: 2  
  IPIN (5,1)  Pin: 1  
  SINK (5,1)  Class: 0  


Net 9 (N12)

SOURCE (4,8)  Pad: 1  
  OPIN (4,8)  Pad: 1  
 CHANX (4,7)  Track: 3  
 CHANY (3,7)  Track: 3  
 CHANY (3,6)  Track: 3  
 CHANY (3,5)  Track: 3  
 CHANY (3,4)  Track: 3  
 CHANY (3,3)  Track: 3  
  IPIN (4,3)  Pin: 3  
  SINK (4,3)  Class: 0  
 CHANY (3,3)  Track: 3  
  IPIN (3,3)  Pin: 1  
  SINK (3,3)  Class: 0  


Net 10 (N13)

SOURCE (1,0)  Pad: 1  
  OPIN (1,0)  Pad: 1  
 CHANX (1,0)  Track: 3  
  IPIN (1,1)  Pin: 2  
  SINK (1,1)  Class: 0  
  OPIN (1,0)  Pad: 1  
 CHANX (1,0)  Track: 0  
 CHANY (1,1)  Track: 0  
  IPIN (2,1)  Pin: 3  
  SINK (2,1)  Class: 0  


Net 11 (N10)

SOURCE (0,3)  Pad: 1  
  OPIN (0,3)  Pad: 1  
 CHANY (0,3)  Track: 3  
 CHANX (1,2)  Track: 2  
  IPIN (1,2)  Pin: 0  
  SINK (1,2)  Class: 0  
 CHANX (1,2)  Track: 2  
 CHANX (2,2)  Track: 2  
  IPIN (2,2)  Pin: 0  
  SINK (2,2)  Class: 0  
 CHANX (2,2)  Track: 2  
 CHANY (2,3)  Track: 2  
  IPIN (3,3)  Pin: 3  
  SINK (3,3)  Class: 0  


Net 12 (N11)

SOURCE (8,2)  Pad: 1  
  OPIN (8,2)  Pad: 1  
 CHANY (7,2)  Track: 2  
 CHANX (7,2)  Track: 3  
  IPIN (7,2)  Pin: 0  
  SINK (7,2)  Class: 0  
 CHANX (7,2)  Track: 3  
 CHANY (6,2)  Track: 3  
 CHANY (6,1)  Track: 3  
  IPIN (7,1)  Pin: 3  
  SINK (7,1)  Class: 0  
 CHANY (6,1)  Track: 3  
  IPIN (6,1)  Pin: 1  
  SINK (6,1)  Class: 0  


Net 13 (N14)

SOURCE (0,6)  Pad: 1  
  OPIN (0,6)  Pad: 1  
 CHANY (0,6)  Track: 1  
 CHANY (0,5)  Track: 1  
 CHANY (0,4)  Track: 1  
 CHANX (1,3)  Track: 0  
  IPIN (1,3)  Pin: 0  
  SINK (1,3)  Class: 0  
 CHANY (0,4)  Track: 1  
 CHANY (0,3)  Track: 1  
 CHANY (0,2)  Track: 1  
  IPIN (1,2)  Pin: 3  
  SINK (1,2)  Class: 0  


Net 14 (N15)

SOURCE (4,0)  Pad: 1  
  OPIN (4,0)  Pad: 1  
 CHANX (4,0)  Track: 3  
  IPIN (4,1)  Pin: 2  
  SINK (4,1)  Class: 0  
 CHANX (4,0)  Track: 3  
 CHANY (3,1)  Track: 2  
 CHANY (3,2)  Track: 2  
  IPIN (4,2)  Pin: 3  
  SINK (4,2)  Class: 0  


Net 15 (N29)

SOURCE (3,3)  Class: 1  
  OPIN (3,3)  Pin: 5  
 CHANY (3,3)  Track: 1  
 CHANY (3,2)  Track: 1  
 CHANY (3,1)  Track: 1  
  IPIN (4,1)  Pin: 3  
  SINK (4,1)  Class: 0  


Net 16 (N28)

SOURCE (2,1)  Class: 1  
  OPIN (2,1)  Pin: 5  
 CHANY (2,1)  Track: 1  
 CHANX (3,0)  Track: 0  
 CHANX (4,0)  Track: 0  
 CHANX (5,0)  Track: 0  
 CHANY (5,1)  Track: 0  
  IPIN (6,1)  Pin: 3  
  SINK (6,1)  Class: 0  


Net 17 (N23)

SOURCE (2,3)  Class: 1  
  OPIN (2,3)  Pin: 7  
 CHANY (1,3)  Track: 2  
  IPIN (1,3)  Pin: 1  
  SINK (1,3)  Class: 0  


Net 18 (N22)

SOURCE (6,2)  Class: 1  
  OPIN (6,2)  Pin: 7  
 CHANY (5,2)  Track: 3  
  IPIN (5,2)  Pin: 1  
  SINK (5,2)  Class: 0  


Net 19 (N21)

SOURCE (5,2)  Class: 1  
  OPIN (5,2)  Pin: 6  
 CHANX (5,1)  Track: 1  
  IPIN (5,1)  Pin: 0  
  SINK (5,1)  Class: 0  


Net 20 (N36)

SOURCE (1,1)  Class: 1  
  OPIN (1,1)  Pin: 7  
 CHANY (0,1)  Track: 2  
  IPIN (0,1)  Pad: 0  
  SINK (0,1)  Pad: 0  


Net 21 (N35)

SOURCE (7,2)  Class: 1  
  OPIN (7,2)  Pin: 4  
 CHANX (7,2)  Track: 2  
 CHANY (7,3)  Track: 2  
 CHANY (7,4)  Track: 2  
  IPIN (8,4)  Pad: 0  
  SINK (8,4)  Pad: 0  


Net 22 (N39)

SOURCE (5,1)  Class: 1  
  OPIN (5,1)  Pin: 6  
 CHANX (5,0)  Track: 3  
  IPIN (5,0)  Pad: 0  
  SINK (5,0)  Pad: 0  


Net 23 (N25)

SOURCE (4,2)  Class: 1  
  OPIN (4,2)  Pin: 7  
 CHANY (3,2)  Track: 3  
  IPIN (3,2)  Pin: 1  
  SINK (3,2)  Class: 0  


Net 24 (N24)

SOURCE (3,2)  Class: 1  
  OPIN (3,2)  Pin: 7  
 CHANY (2,2)  Track: 1  
 CHANX (2,1)  Track: 1  
 CHANX (1,1)  Track: 1  
  IPIN (1,1)  Pin: 0  
  SINK (1,1)  Class: 0  


Net 25 (N16)

SOURCE (3,1)  Class: 1  
  OPIN (3,1)  Pin: 5  
 CHANY (3,1)  Track: 0  
 CHANY (3,2)  Track: 0  
 CHANX (4,2)  Track: 0  
 CHANX (5,2)  Track: 0  
 CHANX (6,2)  Track: 0  
 CHANY (6,2)  Track: 1  
  IPIN (7,2)  Pin: 3  
  SINK (7,2)  Class: 0  


Net 26 (N17)

SOURCE (2,2)  Class: 1  
  OPIN (2,2)  Pin: 5  
 CHANY (2,2)  Track: 0  
 CHANX (3,2)  Track: 0  
 CHANY (3,3)  Track: 0  
 CHANX (4,3)  Track: 0  
 CHANY (4,3)  Track: 1  
 CHANY (4,2)  Track: 1  
 CHANX (4,1)  Track: 1  
 CHANX (3,1)  Track: 1  
  IPIN (3,1)  Pin: 0  
  SINK (3,1)  Class: 0  


Net 27 (N18)

SOURCE (4,1)  Class: 1  
  OPIN (4,1)  Pin: 5  
 CHANY (4,1)  Track: 0  
 CHANX (5,1)  Track: 0  
 CHANX (6,1)  Track: 0  
 CHANX (7,1)  Track: 0  
  IPIN (7,1)  Pin: 0  
  SINK (7,1)  Class: 0  


Net 28 (N37)

SOURCE (1,3)  Class: 1  
  OPIN (1,3)  Pin: 4  
 CHANX (1,3)  Track: 3  
 CHANY (0,4)  Track: 2  
 CHANY (0,5)  Track: 2  
  IPIN (0,5)  Pad: 0  
  SINK (0,5)  Pad: 0  


Net 29 (N38)

SOURCE (7,1)  Class: 1  
  OPIN (7,1)  Pin: 6  
 CHANX (7,0)  Track: 3  
  IPIN (7,0)  Pad: 0  
  SINK (7,0)  Pad: 0  


Net 30 (N34)

SOURCE (6,1)  Class: 1  
  OPIN (6,1)  Pin: 5  
 CHANY (6,1)  Track: 2  
 CHANX (7,1)  Track: 2  
 CHANY (7,1)  Track: 3  
  IPIN (8,1)  Pad: 0  
  SINK (8,1)  Pad: 0  


Net 31 (N32)

SOURCE (4,3)  Class: 1  
  OPIN (4,3)  Pin: 6  
 CHANX (4,2)  Track: 1  
 CHANX (3,2)  Track: 1  
 CHANX (2,2)  Track: 1  
  IPIN (2,3)  Pin: 2  
  SINK (2,3)  Class: 0  


Net 32 (N33)

SOURCE (1,2)  Class: 1  
  OPIN (1,2)  Pin: 5  
 CHANY (1,2)  Track: 1  
 CHANX (2,1)  Track: 0  
  IPIN (2,1)  Pin: 0  
  SINK (2,1)  Class: 0  
